src/utils/video.ts

```diff
@@ -10,3 +10,41 @@ export const formatVideoViewsAndDate = (
   const formattedViews = views !== null && (fullViews ? formatNumber(views) : formatNumberShort(views))
   return formattedViews ? `${formattedViews} views • ${formattedDate}` : formattedDate
 }
+
+export type VideoMetadata = {
+  mimeType: string
+  sizeInBytes: number
+  height: number
+  width: number
+  duration: number
+}
+export const getVideoMetadata = async (
+  file?: File
+): Promise<{ metadata: VideoMetadata; error?: undefined } | { metadata?: undefined; error: Error }> => {
```

Comment:
I don't think we need error as a return value. Promises provide a native `reject` functionality and that can carry the error by itself

Reply:
true, it was mostly so we could destructure `const { metadata, error } = await ...` for easier consuming API, what you think?